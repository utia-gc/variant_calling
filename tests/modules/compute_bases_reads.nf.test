nextflow_process {

    name "Test Process compute_bases_reads"
    script "modules/compute_bases_reads.nf"
    process "compute_bases_reads"

    test("compute_bases_reads computes the number of bases in reads -- SE reads; one read") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw"
                    ],
                    [
                        file("${readsLane1.getR1()}"),
                    ],
                    [
                        file("${readsLane1.getR2()}"),
                    ]
                ]
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test expected number bases
            with(process.out.bases.get(0)) {
                assert size() == 2
                // test metadata
                with(get(0)) {
                    assert sampleName == "${readsLane1.getSampleName()}"
                    assert readType   == "${readsLane1.getReadType()}"
                    assert trimStatus == "raw"
                }
                // test number of bases
                with(get(1)) {
                    assert isNumber()
                    assert toInteger() > 0
                }
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }


    test("compute_bases_reads computes the number of bases in reads -- SE reads; multiple reads") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()
        def readsLane2 = new ReadsSELane2()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw"
                    ],
                    [
                        file("${readsLane1.getR1()}"),
                        file("${readsLane2.getR1()}"),
                    ],
                    [
                        file("${readsLane1.getR2()}"),
                        file("${readsLane2.getR2()}"),
                    ]
                ]
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test expected number bases
            with(process.out.bases.get(0)) {
                assert size() == 2
                // test metadata
                with(get(0)) {
                    assert sampleName == "${readsLane1.getSampleName()}"
                    assert readType   == "${readsLane1.getReadType()}"
                    assert trimStatus == "raw"
                }
                // test number of bases
                with(get(1)) {
                    assert isNumber()
                    assert toInteger() > 0
                }
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }


    test("compute_bases_reads computes the number of bases in reads -- PE reads; one read") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw"
                    ],
                    [
                        file("${readsLane1.getR1()}"),
                    ],
                    [
                        file("${readsLane1.getR2()}"),
                    ]
                ]
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test expected number bases
            with(process.out.bases.get(0)) {
                assert size() == 2
                // test metadata
                with(get(0)) {
                    assert sampleName == "${readsLane1.getSampleName()}"
                    assert readType   == "${readsLane1.getReadType()}"
                    assert trimStatus == "raw"
                }
                // test number of bases
                with(get(1)) {
                    assert isNumber()
                    assert toInteger() > 0
                }
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }


    test("compute_bases_reads computes the number of bases in reads -- PE reads; multiple reads") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()
        def readsLane2 = new ReadsPELane2()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw"
                    ],
                    [
                        file("${readsLane1.getR1()}"),
                        file("${readsLane2.getR1()}"),
                    ],
                    [
                        file("${readsLane1.getR2()}"),
                        file("${readsLane2.getR2()}"),
                    ]
                ]
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test expected number bases
            with(process.out.bases.get(0)) {
                assert size() == 2
                // test metadata
                with(get(0)) {
                    assert sampleName == "${readsLane1.getSampleName()}"
                    assert readType   == "${readsLane1.getReadType()}"
                    assert trimStatus == "raw"
                }
                // test number of bases
                with(get(1)) {
                    assert isNumber()
                    assert toInteger() > 0
                }
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }


}
