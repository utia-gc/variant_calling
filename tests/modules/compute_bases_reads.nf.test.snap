{
    "compute_bases_reads computes the number of bases in reads -- PE reads; multiple reads": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "trimStatus": "raw"
                        },
                        "15098974"
                    ]
                ],
                "bases": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "trimStatus": "raw"
                        },
                        "15098974"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-22T10:57:37.368338929"
    },
    "compute_bases_reads computes the number of bases in reads -- SE reads; one read": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "trimStatus": "raw"
                        },
                        "2500000"
                    ]
                ],
                "bases": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "trimStatus": "raw"
                        },
                        "2500000"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-22T10:54:49.824916851"
    },
    "compute_bases_reads computes the number of bases in reads -- SE reads; multiple reads": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "trimStatus": "raw"
                        },
                        "5000000"
                    ]
                ],
                "bases": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "trimStatus": "raw"
                        },
                        "5000000"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-22T10:57:13.298043403"
    },
    "compute_bases_reads computes the number of bases in reads -- PE reads; one read": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "trimStatus": "raw"
                        },
                        "7549612"
                    ]
                ],
                "bases": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "trimStatus": "raw"
                        },
                        "7549612"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-22T10:57:25.2137076"
    }
}